src/main/java/org/prgrms/kdt/customer/BlackCustomerManager.java

```diff
@@ -0,0 +1,59 @@
+package org.prgrms.kdt.customer;
+
+import org.prgrms.kdt.voucher.FileVoucherManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Repository;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+
+@Repository
+public class BlackCustomerManager {
+
+    private static Logger logger = LoggerFactory.getLogger(FileVoucherManager.class);
+
+    private static final String FILE_PATH = "src/main/resources/customer_blacklist.csv";
+
+    private final File blacklistCsv;
+
+    public BlackCustomerManager() {
+        this.blacklistCsv = createFile();
+    }
+
+    private static File createFile() {
+        File file = new File(FILE_PATH);
+        validateFile(file);
+        return file;
+    }
+
+    private static void validateFile(File file) {
+        if (!file.exists()) {
+            createFile(file);
+        }
+    }
+
+    private static void createFile(File file) {
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
+            logger.error("Cannot create file. Please check file name or path.");
+            throw new RuntimeException("File Create Error");
+        }
+    }
+
+    public List<Customer> findAll() {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(blacklistCsv))) {
+            return bufferedReader.lines()
+                    .map(line -> line.split(", "))
+                    .map(t -> new Customer(t[0], t[1]))
+                    .toList();
+        } catch (IOException e) {
+            logger.error("Cannot find file. Please check there is file those name is {}", blacklistCsv.getName());
+            throw new RuntimeException("File Read Error");
```

Comment:
보통 Exception 클래스들은 생성자에 메시지와 함께 `Throwable` 타입의 인스턴스를 받을 수 있도록 해놨어요.
예외가 발생했을 때 생성자에 메시지만 넘기는 것과 `Throwable` 타입의 인스턴스를 함께 넘길때 로그가 어떻게 남는지 한 번 확인 해보면 좋겠네요

- e 보다는 의미있는 이름을 지어주면 좋을 것 같아요

```suggestion
            throw new RuntimeException("File Read Error", exception);
```

Reply:
Throwable을 넣어주면, catch를 하는 곳에서 Throwable 를 쓸 수 있겠네요! 
만약에 넣어주지 않으면 메시지만 조회할 수 있겠네요...
음 그럼 대안으로 알려주신 exception을 써보려고 하는데 이것은 어떤가요?

Reply:
> Throwable을 넣어주면, catch를 하는 곳에서 Throwable 를 쓸 수 있겠네요! 만약에 넣어주지 않으면 메시지만 조회할 수 있겠네요... 음 그럼 대안으로 알려주신 exception을 써보려고 하는데 이것은 어떤가요?

대안은 아니고 `RuntimeException`을 따라 올라가다보면 `Throwable`이 나올거에요. 그러니 지금 `RuntimeException`을 인자로 넘겨도 아무런 지정이 없습니다!

사실 제가 의도한 이야기는 만약 예외가 발생했을 때 `Throwable`을 넘겨주지 않으면 예외 로그가 누적되지 않아서 어디서부터 예외가 발생했는지 보이질 않아요. 이거는 말로 설명드리기 어려우니 나중이 직접 보여 드릴 시간을 가지면 좋을 것 같긴 하네요

Reply:
음 그렇군요🤔 throwable이 누적되서 나중에 디버깅을 쉽게 할 때 사용하나 보네요.. 나중에 게더에서 찾아뵙겠습니다...!